src/index.ts

```diff
@@ -57,6 +60,8 @@ ${chalk.bold('Options:')}
     --clean             Clear out the destination directory before install.
     --optimize          Transpile, minify, and optimize installed dependencies for production.
     --babel             Transpile installed dependencies. Also enabled with "--optimize".
+    --entry             Entry file(s) to recursively scan for imports, comma-separated
```

Comment:
does this support globs? If so, mention that here. 

This may also make more sense as a package.json config value, alongside "webDependencies". Eventually we want everything to be configurable by that JSON config, but arrays are especially confusing to pass via CLI arg so I think this may make more sense as a JSON config to start.

Reply:
Up in the PR comment under “Features”:
> **No glob support**: no, really—this is a feature! In order for tree-shaking to work properly, we don’t want to support `--entry src/**/*.js`—that will bundle too much, and could be reading files it doesn’t need to be reading. Instead, we want to manually specify all entry files so only those things used will get bundled.

Reply:
ooooh interesting. Sorry i missed that. So the idea is that we point to single entrypoints, but then local imports are crawled from those entrypoints as well? 

One nice thing about serving code unbundled is that if an import statement is never read, the dep is never loaded. So you're right to point out that a dep may be installed unnecessarily, but it would never be loaded by the browser unnecessarily. 

I have a feeling this will end up being something that users get stuck on / is not intuitive to how they would expect it to work (in the same way I misunderstood). But if you feel strongly, I'm willing to try it out for this first version, as long as you're okay with us choosing to add glob support later on down the road in a follow up PR if needed?

Reply:
Exactly. You’re correct that it probably won’t impact client weight. The major bottleneck I was concerned about was Babel’s parse time—scanning files that were ultimately unused. I don’t have metrics; was more of an aim to make it as fast as possible initially.

But like you said “entry” is less intuitive than a glob, especially if you have 1 per page. And even if we did glob, it won’t ever get stuck pinballing around node_modules so it probably won’t be too bad.

I’ll think on this, but maybe glob would be an improvement. I’d also like to find a way to parallelize it so that we get globs racing to parse the same files while sharing a cache—that’d be pretty great.

Reply:
Resolving this convo to clean it up. As stated elsewhere globs are now allowed, and the dependency tree should probably just appear as full-fledged stats + bundle analyzer further down the line (if that’s desired)